src/Adapter/Cart/QueryHandler/GetCartInformationHandler.php

```diff
@@ -238,4 +271,125 @@ private function fetchCartDeliveryOptions(array $deliveryOptionsByAddress, int $
         //make sure array is not associative
         return array_values($deliveryOptions);
     }
+
+    /**
+     * @param array $legacySummary
+     * @param Currency $currency
+     * @param Cart $cart
+     *
+     * @return CartSummary
+     *
+     * @throws LocalizationException
+     */
+    private function extractSummaryFromLegacySummary(array $legacySummary, Currency $currency, Cart $cart): CartSummary
+    {
+        $cartId = (int) $cart->id;
+
+        if (0 !== (int) $legacySummary['total_discounts_tax_exc']) {
```

Comment:
The `if/else` is useless here `-1 * 0 == 0` so you can just use `$discount = $this->locale->formatPrice(-1 * $legacySummary['total_discounts_tax_exc'], $currency->iso_code);` for all cases